diff -urN comm-release.orig/configure comm-release/configure
--- comm-release.orig/configure	2011-10-28 14:16:38.731333224 +0400
+++ comm-release/configure	2011-10-28 16:20:42.612261219 +0400
@@ -17938,29 +17938,29 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig""... $ac_c" 1>&6
-echo "configure:17943: checking for cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" >&5
+        echo $ac_n "checking for cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig""... $ac_c" 1>&6
+echo "configure:17943: checking for cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" >&5
 
-        if $PKG_CONFIG --exists "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" ; then
+        if $PKG_CONFIG --exists "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking CAIRO_CFLAGS""... $ac_c" 1>&6
 echo "configure:17950: checking CAIRO_CFLAGS" >&5
-            CAIRO_CFLAGS=`$PKG_CONFIG --cflags "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
+            CAIRO_CFLAGS=`$PKG_CONFIG --cflags "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
             echo "$ac_t""$CAIRO_CFLAGS" 1>&6
 
             echo $ac_n "checking CAIRO_LIBS""... $ac_c" 1>&6
 echo "configure:17955: checking CAIRO_LIBS" >&5
             ## Remove evil flags like -Wl,--export-dynamic
-            CAIRO_LIBS="`$PKG_CONFIG --libs \"cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig\" |sed s/-Wl,--export-dynamic//g`"
+            CAIRO_LIBS="`$PKG_CONFIG --libs \"cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig\" |sed s/-Wl,--export-dynamic//g`"
             echo "$ac_t""$CAIRO_LIBS" 1>&6
         else
             CAIRO_CFLAGS=""
             CAIRO_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            CAIRO_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
+            CAIRO_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
             echo $CAIRO_PKG_ERRORS
         fi
 
@@ -17976,7 +17976,7 @@
      :
   else
      if test "$COMPILE_ENVIRONMENT"; then 
-       { echo "configure: error: Library requirements (cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
+       { echo "configure: error: Library requirements (cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
      fi
   fi
 
@@ -18031,29 +18031,29 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for cairo-xlib-xrender >= $CAIRO_VERSION""... $ac_c" 1>&6
-echo "configure:18036: checking for cairo-xlib-xrender >= $CAIRO_VERSION" >&5
+        echo $ac_n "checking for cairo10-xlib-xrender >= $CAIRO_VERSION""... $ac_c" 1>&6
+echo "configure:18036: checking for cairo10-xlib-xrender >= $CAIRO_VERSION" >&5
 
-        if $PKG_CONFIG --exists "cairo-xlib-xrender >= $CAIRO_VERSION" ; then
+        if $PKG_CONFIG --exists "cairo10-xlib-xrender >= $CAIRO_VERSION" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking CAIRO_XRENDER_CFLAGS""... $ac_c" 1>&6
 echo "configure:18043: checking CAIRO_XRENDER_CFLAGS" >&5
-            CAIRO_XRENDER_CFLAGS=`$PKG_CONFIG --cflags "cairo-xlib-xrender >= $CAIRO_VERSION"`
+            CAIRO_XRENDER_CFLAGS=`$PKG_CONFIG --cflags "cairo10-xlib-xrender >= $CAIRO_VERSION"`
             echo "$ac_t""$CAIRO_XRENDER_CFLAGS" 1>&6
 
             echo $ac_n "checking CAIRO_XRENDER_LIBS""... $ac_c" 1>&6
 echo "configure:18048: checking CAIRO_XRENDER_LIBS" >&5
             ## Remove evil flags like -Wl,--export-dynamic
-            CAIRO_XRENDER_LIBS="`$PKG_CONFIG --libs \"cairo-xlib-xrender >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
+            CAIRO_XRENDER_LIBS="`$PKG_CONFIG --libs \"cairo10-xlib-xrender >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
             echo "$ac_t""$CAIRO_XRENDER_LIBS" 1>&6
         else
             CAIRO_XRENDER_CFLAGS=""
             CAIRO_XRENDER_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            CAIRO_XRENDER_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo-xlib-xrender >= $CAIRO_VERSION"`
+            CAIRO_XRENDER_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo10-xlib-xrender >= $CAIRO_VERSION"`
             echo $CAIRO_XRENDER_PKG_ERRORS
         fi
 
@@ -18069,7 +18069,7 @@
      :
   else
      if test "$COMPILE_ENVIRONMENT"; then 
-       { echo "configure: error: Library requirements (cairo-xlib-xrender >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
+       { echo "configure: error: Library requirements (cairo10-xlib-xrender >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
      fi
   fi
 
diff -urN comm-release.orig/configure.in comm-release/configure.in
--- comm-release.orig/configure.in	2011-10-28 14:16:37.781315736 +0400
+++ comm-release/configure.in	2011-10-28 16:20:42.617261317 +0400
@@ -6664,10 +6664,10 @@
         MOZ_CAIRO_LIBS="$MOZ_CAIRO_LIBS $XLDFLAGS -lXrender -lfreetype -lfontconfig"
     fi
 else
-   PKG_CHECK_MODULES(CAIRO, cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig)
+   PKG_CHECK_MODULES(CAIRO, cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig)
    MOZ_CAIRO_LIBS=$CAIRO_LIBS
    if test "$MOZ_X11"; then
-        PKG_CHECK_MODULES(CAIRO_XRENDER, cairo-xlib-xrender >= $CAIRO_VERSION)
+        PKG_CHECK_MODULES(CAIRO_XRENDER, cairo10-xlib-xrender >= $CAIRO_VERSION)
         MOZ_CAIRO_LIBS="$MOZ_CAIRO_LIBS $XLDFLAGS $CAIRO_XRENDER_LIBS"
    fi
 fi
diff -urN comm-release.orig/mozilla/configure comm-release/mozilla/configure
--- comm-release.orig/mozilla/configure	2011-10-28 14:16:44.912447006 +0400
+++ comm-release/mozilla/configure	2011-10-28 16:24:41.989956177 +0400
@@ -24369,29 +24369,29 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig""... $ac_c" 1>&6
-echo "configure:24374: checking for cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" >&5
+        echo $ac_n "checking for cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig""... $ac_c" 1>&6
+echo "configure:24374: checking for cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" >&5
 
-        if $PKG_CONFIG --exists "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" ; then
+        if $PKG_CONFIG --exists "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking CAIRO_CFLAGS""... $ac_c" 1>&6
 echo "configure:24381: checking CAIRO_CFLAGS" >&5
-            CAIRO_CFLAGS=`$PKG_CONFIG --cflags "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
+            CAIRO_CFLAGS=`$PKG_CONFIG --cflags "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
             echo "$ac_t""$CAIRO_CFLAGS" 1>&6
 
             echo $ac_n "checking CAIRO_LIBS""... $ac_c" 1>&6
 echo "configure:24386: checking CAIRO_LIBS" >&5
             ## Remove evil flags like -Wl,--export-dynamic
-            CAIRO_LIBS="`$PKG_CONFIG --libs \"cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig\" |sed s/-Wl,--export-dynamic//g`"
+            CAIRO_LIBS="`$PKG_CONFIG --libs \"cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig\" |sed s/-Wl,--export-dynamic//g`"
             echo "$ac_t""$CAIRO_LIBS" 1>&6
         else
             CAIRO_CFLAGS=""
             CAIRO_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            CAIRO_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
+            CAIRO_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig"`
             echo $CAIRO_PKG_ERRORS
         fi
 
@@ -24407,7 +24407,7 @@
      :
   else
      if test "$COMPILE_ENVIRONMENT"; then 
-       { echo "configure: error: Library requirements (cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
+       { echo "configure: error: Library requirements (cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
      fi
   fi
 
@@ -24462,29 +24462,29 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for cairo-tee >= $CAIRO_VERSION""... $ac_c" 1>&6
-echo "configure:24467: checking for cairo-tee >= $CAIRO_VERSION" >&5
+        echo $ac_n "checking for cairo10-tee >= $CAIRO_VERSION""... $ac_c" 1>&6
+echo "configure:24467: checking for cairo10-tee >= $CAIRO_VERSION" >&5
 
-        if $PKG_CONFIG --exists "cairo-tee >= $CAIRO_VERSION" ; then
+        if $PKG_CONFIG --exists "cairo10-tee >= $CAIRO_VERSION" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking CAIRO_TEE_CFLAGS""... $ac_c" 1>&6
 echo "configure:24474: checking CAIRO_TEE_CFLAGS" >&5
-            CAIRO_TEE_CFLAGS=`$PKG_CONFIG --cflags "cairo-tee >= $CAIRO_VERSION"`
+            CAIRO_TEE_CFLAGS=`$PKG_CONFIG --cflags "cairo10-tee >= $CAIRO_VERSION"`
             echo "$ac_t""$CAIRO_TEE_CFLAGS" 1>&6
 
             echo $ac_n "checking CAIRO_TEE_LIBS""... $ac_c" 1>&6
 echo "configure:24479: checking CAIRO_TEE_LIBS" >&5
             ## Remove evil flags like -Wl,--export-dynamic
-            CAIRO_TEE_LIBS="`$PKG_CONFIG --libs \"cairo-tee >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
+            CAIRO_TEE_LIBS="`$PKG_CONFIG --libs \"cairo10-tee >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
             echo "$ac_t""$CAIRO_TEE_LIBS" 1>&6
         else
             CAIRO_TEE_CFLAGS=""
             CAIRO_TEE_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            CAIRO_TEE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo-tee >= $CAIRO_VERSION"`
+            CAIRO_TEE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo10-tee >= $CAIRO_VERSION"`
             echo $CAIRO_TEE_PKG_ERRORS
         fi
 
@@ -24500,7 +24500,7 @@
      :
   else
      if test "$COMPILE_ENVIRONMENT"; then 
-       { echo "configure: error: Library requirements (cairo-tee >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
+       { echo "configure: error: Library requirements (cairo10-tee >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
      fi
   fi
 
@@ -24554,29 +24554,29 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo $ac_n "checking for cairo-xlib-xrender >= $CAIRO_VERSION""... $ac_c" 1>&6
-echo "configure:24559: checking for cairo-xlib-xrender >= $CAIRO_VERSION" >&5
+        echo $ac_n "checking for cairo10-xlib-xrender >= $CAIRO_VERSION""... $ac_c" 1>&6
+echo "configure:24559: checking for cairo10-xlib-xrender >= $CAIRO_VERSION" >&5
 
-        if $PKG_CONFIG --exists "cairo-xlib-xrender >= $CAIRO_VERSION" ; then
+        if $PKG_CONFIG --exists "cairo10-xlib-xrender >= $CAIRO_VERSION" ; then
             echo "$ac_t""yes" 1>&6
             succeeded=yes
 
             echo $ac_n "checking CAIRO_XRENDER_CFLAGS""... $ac_c" 1>&6
 echo "configure:24566: checking CAIRO_XRENDER_CFLAGS" >&5
-            CAIRO_XRENDER_CFLAGS=`$PKG_CONFIG --cflags "cairo-xlib-xrender >= $CAIRO_VERSION"`
+            CAIRO_XRENDER_CFLAGS=`$PKG_CONFIG --cflags "cairo10-xlib-xrender >= $CAIRO_VERSION"`
             echo "$ac_t""$CAIRO_XRENDER_CFLAGS" 1>&6
 
             echo $ac_n "checking CAIRO_XRENDER_LIBS""... $ac_c" 1>&6
 echo "configure:24571: checking CAIRO_XRENDER_LIBS" >&5
             ## Remove evil flags like -Wl,--export-dynamic
-            CAIRO_XRENDER_LIBS="`$PKG_CONFIG --libs \"cairo-xlib-xrender >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
+            CAIRO_XRENDER_LIBS="`$PKG_CONFIG --libs \"cairo10-xlib-xrender >= $CAIRO_VERSION\" |sed s/-Wl,--export-dynamic//g`"
             echo "$ac_t""$CAIRO_XRENDER_LIBS" 1>&6
         else
             CAIRO_XRENDER_CFLAGS=""
             CAIRO_XRENDER_LIBS=""
             ## If we have a custom action on failure, don't print errors, but 
             ## do set a variable so people can do so.
-            CAIRO_XRENDER_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo-xlib-xrender >= $CAIRO_VERSION"`
+            CAIRO_XRENDER_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "cairo10-xlib-xrender >= $CAIRO_VERSION"`
             echo $CAIRO_XRENDER_PKG_ERRORS
         fi
 
@@ -24592,7 +24592,7 @@
      :
   else
      if test "$COMPILE_ENVIRONMENT"; then 
-       { echo "configure: error: Library requirements (cairo-xlib-xrender >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
+       { echo "configure: error: Library requirements (cairo10-xlib-xrender >= $CAIRO_VERSION) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." 1>&2; exit 1; }
      fi
   fi
 
diff -urN comm-release.orig/mozilla/configure.in comm-release/mozilla/configure.in
--- comm-release.orig/mozilla/configure.in	2011-10-28 14:16:39.054339169 +0400
+++ comm-release/mozilla/configure.in	2011-10-28 16:25:04.612399874 +0400
@@ -8603,12 +8603,12 @@
     mv -f $CAIRO_FEATURES_H "$CAIRO_FEATURES_H".orig 2> /dev/null
 
 else
-   PKG_CHECK_MODULES(CAIRO, cairo >= $CAIRO_VERSION pixman-1 freetype2 fontconfig)
+   PKG_CHECK_MODULES(CAIRO, cairo10 >= $CAIRO_VERSION pixman-1 freetype2 fontconfig)
    MOZ_CAIRO_CFLAGS=$CAIRO_CFLAGS
    MOZ_CAIRO_LIBS=$CAIRO_LIBS
-   PKG_CHECK_MODULES(CAIRO_TEE, cairo-tee >= $CAIRO_VERSION)
+   PKG_CHECK_MODULES(CAIRO_TEE, cairo10-tee >= $CAIRO_VERSION)
    if test "$MOZ_X11"; then
-        PKG_CHECK_MODULES(CAIRO_XRENDER, cairo-xlib-xrender >= $CAIRO_VERSION)
+        PKG_CHECK_MODULES(CAIRO_XRENDER, cairo10-xlib-xrender >= $CAIRO_VERSION)
         MOZ_CAIRO_LIBS="$MOZ_CAIRO_LIBS $XLDFLAGS $CAIRO_XRENDER_LIBS"
         MOZ_CAIRO_CFLAGS="$MOZ_CAIRO_CFLAGS $CAIRO_XRENDER_CFLAGS"
    fi
